src/player.cpp

```diff
@@ -2097,11 +2097,30 @@ Item* Player::getCorpse(Creature* lastHitCreature, Creature* mostDamageCreature)
 {
 	Item* corpse = Creature::getCorpse(lastHitCreature, mostDamageCreature);
 	if (corpse && corpse->getContainer()) {
+		std::map<std::string, uint16_t> names;
+		for (const auto& killer : getKillers()) {
```

Comment:
If what @EPuncker mentions is true then his suggestion cannot be a solution, however this means that the current code must be modified, I will try to improve the structure of the code a bit so that it fits the tests.